SigAverager/testing/test_dsp_signal_averager.py

```diff
@@ -0,0 +1,128 @@
+"""Unit test for signal averager."""
+import SigAverager
+import numpy as np
+import logging
+import matplotlib.pyplot as plt
+
+
+def test_signal_averaging():
+    """
+    Test signal averaging.
+
+    Test Overview:
+    --------------
+    This test will pass parametets to signal averaging method and get back a vector.
+    The intention is to check for the correct averaging. The SNR should improve with the number of iterations.
+    The test CW tone is 75MHz (EE3:75).
+    The test will look for the following:
+    a) Is the SNR correct? This is verified using an FFT and computing the SNR.
+
+    Parameters
+    ----------
+    cw_scale: float
+        factor to scale generated noise.
+    cw_freq: float
+        Frequency of CW to be generated.
+    sampling_frequency: int
+        Sample rate for generated CW. This is expressed in Hz. E.g. 1712e6.
+    num_samples: int
+        Number of samples for generated CW.
+    noise_scale: float
+        Factor to scale generated noise.
+    n_iter: int
+        Number of iterations to average signal.
+    expected_SNR_dB: int
+        Expected SNR after signal averaging.
```

Comment:
So generally the "Parameters" heading refers to the arguments that you pass to the function. In this case you don't have `test_signal_averaging(cw_scale, cw_freq, ...)`, but you are setting these variables inside the function.

Reply:
Noted.

Reply:
If you do want to document these things however, you can use the heading “Attributes”